src/js/player.js

```diff
@@ -1873,12 +1873,34 @@ class Player extends Component {
    * Attempt to begin playback at the first opportunity.
    *
    * @return {Promise|undefined}
-   *         Returns a `Promise` only if the browser returns one and the player
-   *         is ready to begin playback. For some browsers and all non-ready
-   *         situations, this will return `undefined`.
+   *         Returns a promise if the browser supports Promises (or one
+   *         was passed in as an option). This promise will be resolved on
+   *         the return value of play. If this is undefined it will fulfill the
+   *         promise chain otherwise the promise chain will be fulfilled when
+   *         the promise from play is fulfilled.
    */
   play() {
+    const promise = this.options_.Promise || window.Promise;
 
+    if (promise) {
+      return new Promise((resolve) => {
```

Comment:
I wonder if it's possible to have `play_` return a promise itself or use a promise we give it. Maybe a bit tricky since our `play_` method has some event listeners.

Reply:
whats really tricky is that IE 11 does not have promise support, so we can't technically return a promise all the time.

Reply:
yeah, IE11 aside, I mean.
We can require a polyfill for the best behavior here, as long as we don't fail if one doesn't provided.